bait-files/src/test/java/com/google/sps/UserTest.java

```diff
@@ -0,0 +1,131 @@
+package com.google.sps;
+
+import static com.google.appengine.api.datastore.FetchOptions.Builder.withLimit;
+import static com.google.sps.data.Keys.*;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+import static org.mockito.Mockito.*;
+
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.After;
+import org.junit.Test;
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.users.UserService;
+import com.google.appengine.api.users.UserServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.tools.development.testing.LocalServiceTestHelper;
+import com.google.appengine.tools.development.testing.LocalUserServiceTestConfig;
+import java.io.*;
+import javax.servlet.http.*;
+import javax.servlet.ServletException;
+import com.google.sps.servlets.UsernameTakenServlet;
+import com.google.sps.servlets.LoginServlet;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+@RunWith(JUnit4.class)
+public final class UserTest {
+
+  private final LocalServiceTestHelper helper =
+      new LocalServiceTestHelper(new LocalUserServiceTestConfig())
+          .setEnvIsAdmin(true).setEnvIsLoggedIn(true).setEnvEmail("andrew").setEnvAuthDomain("gmail.com");;
+
+  @Before
+  public void setUp() {
+    helper.setUp();
+  }
+
+  @After
+  public void tearDown() {
+    helper.tearDown();
+  }
+
+  // Database is empty, so username should not be taken
+  @Test
+  public void emptyUsernameTakenTest() throws IOException {
```

Comment:
So a note about unit test naming...  There are a lot of conventions, but for the most part what we do on our team is to follow this naming convention:
public void methodBeingTested_setupInfoForThisTest_expectedResult()

So in this first case, I'd go with:
public void doGet_usernameNotInDatabase_returnsNotTaken

Or something like that.  Now when people read the test name, they can see that this is a test of the doGet method, for the case when the proposed username does not already exist in the database, and the expected result is that the response will say that the username is not taken =)

This is a pretty concise and to the point way of writing test names, and it also means that we don't need to write a comment before every test, so you'll be able to delete those.  As in other situations, it's always nice when the names of things in code are clear enough that comments aren't required =)

Also if you're wondering, we do normally avoid using underscores in naming methods and variables (other than constant variables, which are in all uppercase), but the naming of test methods is a specific exception when we very much do use them.

Reply:
Changed name